packages/zoe/src/zoeService/zoeSeat.js

```diff
@@ -35,8 +35,17 @@ export const makeZoeSeatAdminKit = (
   // This does not suppress any error messages.
   payoutPromiseKit.promise.catch(_ => {});
   const { notifier, updater } = makeNotifierKit();
-
-  let currentAllocation = initialAllocation;
+  let actualProposal;
+  let proposalSupplied;
+  if (!proposal) {
+    proposalSupplied = false;
```

Comment:
Whoops, `proposalSupplied` needs to be a function so that it gets the latest assignment of `proposal`. Other than that, I think this is still a helpful change, though.